server/src/main/java/org/elasticsearch/bootstrap/BootstrapChecks.java

```diff
@@ -144,19 +144,19 @@ static void check(
         for (final BootstrapCheck check : checks) {
             final BootstrapCheck.BootstrapCheckResult result = check.check(context);
             if (result.isFailure()) {
-                if (!(enforceLimits || enforceBootstrapChecks) && !check.alwaysEnforce()) {
+                if ((enforceLimits || enforceBootstrapChecks) && !check.alwaysEnforce() == false) {
```

Comment:
This could be either:

```
((enforceLimits || enforceBootstrapChecks || check.alwaysEnforce()) == false)
```

or…

```
(enforceLimits == false && enforceBootstrapChecks == false && check.alwaysEnforce() == false)
```

Reply:
Good catch, thank you 🙏